share/gpodder/extensions/sonos.py

```diff
@@ -25,38 +25,35 @@
 
 SONOS_CAN_PLAY = lambda e: 'audio' in e.file_type()
 
-
 class gPodderExtension:
     def __init__(self, container):
         speakers = soco.discover()
         logger.info('Found Sonos speakers: %s' % ', '.join(name.player_name for name in speakers))
 
         self.speakers = {}
-        for controller in speakers:
+        for speaker in speakers:
 
             try:
-                info = controller.get_speaker_info()
+                info = speaker.get_speaker_info()
 
             except requests.ConnectionError as ce:
                 # ignore speakers we can't connect to
                 continue
 
             name = info.get('zone_name', None)
+            uid = speaker.uid
 
             # devices that do not have a name are probably bridges
             if name:
-                self.speakers[speaker_ip] = name
-                speaker_ip = controller.ip_address
+                self.speakers[speaker.uid] = speaker
 
-    def _stream_to_speaker(self, speaker_ip, episodes):
+    def _stream_to_speaker(self, speaker_uid, episodes):
         """ Play or enqueue selected episodes """
 
         urls = [episode.url for episode in episodes if SONOS_CAN_PLAY(episode)]
-        logger.info('Streaming to Sonos %s: %s' % (speaker_ip, ', '.join(urls)))
+        logger.info('Streaming to Sonos %s: %s'%(self.speakers[speaker_uid].ip_address, ', '.join(urls)))
```

Comment:
add space before and after modulo operator (pycodestyle)

Reply:
Well, technically, modulo's shouldn't be used anymore from Python 3; `.format` [should be](https://docs.python.org/2/library/stdtypes.html#str.format).  Fixed though.